olp-cpp-sdk-dataservice-read/src/repositories/NamedMutex.cpp

```diff
@@ -145,7 +145,10 @@ NamedMutex::NamedMutex(NamedMutexStorage& storage, const std::string& name,
 NamedMutex::~NamedMutex() { storage_.ReleaseLock(name_); }
 
 void NamedMutex::lock() {
-  if (!context_.IsCancelled() && !try_lock()) {
+  const bool valid = context_.ExecuteOrCancelled(
+      [&]() { return client::CancellationToken([&]() { Notify(); }); });
```

Comment:
It looks dangerous that 'this' is captured in lambda of object witch lifetime is bigger than NamedMutex. I would either overwrite lambda when 'unlock' is near finish or i would make Notify static and of necessary data outside